tools/install_ubuntu_docs_dependencies.sh

```diff
@@ -2,10 +2,30 @@
 #
 # Prepare an Ubuntu CI machine for running 'tox -e docs'.  Assumes that Python is available.
 
-set -e
+if [ $GITHUB_TOKEN ]; then
+    set -e
 
-python -m pip install --upgrade pip setuptools wheel
-python -m pip install --upgrade tox
+    python -m pip install --upgrade pip setuptools wheel
+    python -m pip install --upgrade tox
 
-sudo apt-get update
-sudo apt-get install -y graphviz pandoc doxygen
+    sudo apt-get update
+    sudo apt-get install -y graphviz pandoc 
+
+    # This command fetches the latest release of doxygen and its linux binaries
+    curl -H "Authorization: token $GITHUB_TOKEN" \
+        https://api.github.com/repos/doxygen/doxygen/releases/latest | \
+        jq '.["assets"][2]["browser_download_url"]' | \
+        xargs -I {} wget {} -O ./doxygen.tar.gz
```

Comment:
Using `.["assets"][2]["browser_download_url"]` is assuming that the order the items come out in the release will always be the same, which is fragile.  But we probably don't want to attempt to get the "latest" anyway - it's more reliable to stick with a known-good version.